src/sql/run/sparkdataframe.py

```
@@ -1,19 +1,27 @@
 try:
     from pyspark.sql import DataFrame
     from pyspark.sql.connect.dataframe import DataFrame as CDataFrame
+    from pyspark.sql.utils import AnalysisException
 except ModuleNotFoundError:
     DataFrame = None
     CDataFrame = None
+    AnalysisException = None
 
 from sql import exceptions
 
 
 def handle_spark_dataframe(dataframe, should_cache=False):
     """Execute a ResultSet sqlaproxy using pysark module."""
-    if not DataFrame and not CDataFrame:
+    if not DataFrame and not CDataFrame and not AnalysisException:
         raise exceptions.MissingPackageError("pysark not installed")
 
-    return SparkResultProxy(dataframe, dataframe.columns, should_cache)
+    try:
+        return SparkResultProxy(dataframe, dataframe.columns, should_cache)
+    except AnalysisException as e:
```

this except is redundant, the `except Exception as e` can catch all exceptions